jamazon/main.js

```diff
@@ -268,9 +386,26 @@ $cart.addEventListener('click', function (event) {
   if (event.target.getAttribute('id') === 'returncart') {
     app.view = 'catalog'
   }
+  else if (event.target.getAttribute('id') === 'checkout') {
+    app.view = 'checkout'
+    checkout(app.cart)
+  }
+  renderAll()
+})
+
+$siteHead.addEventListener('click', function (event) {
+  if (event.target.getAttribute('id') === 'header') {
+    app.view = 'catalog'
+  }
   renderAll()
 })
 
+$checkout.addEventListener('click', function (event) {
```

Comment:
In general, we listen for `'submit'` events when dealing with forms, not clicks. But be sure to prevent the default behavior of form submissions.